torch/csrc/api/include/torch/nn/functional/activation.h

```diff
@@ -48,6 +48,27 @@ inline Tensor logsigmoid(const Tensor& input) {
   return torch::log_sigmoid(input);
 }
 
+inline Tensor gumbel_softmax(const Tensor& logits, const GumbelSoftmaxOptions& options = {}) {
+    if (options.eps() != 1e-10) {
+        std::cerr << "`eps` parameter is deprecated and has no effect." << std::endl;
+    }
+
+    torch::Tensor gumbels = -torch::empty_like(logits).exponential_().log();  // ~Gumbel(0,1)
+    gumbels = (logits + gumbels) / options.tau();  // ~Gumbel(logits, tau)
+    torch::Tensor y_soft = gumbels.softmax(options.dim());
+
+    if (options.hard()) {
+        // Straight through.
+        torch::Tensor index = std::get<1>(y_soft.max(options.dim(), true));
```

Comment:
same here: `torch::Tensor` -> `auto`

Reply:
it would also be awesome to write:
```suggestion
        auto index = std::get<1>(y_soft.max(options.dim(), /*keepdim=*/true));
```
to make it more clear the purpose of the second argument :D